common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -2465,6 +2465,12 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "If the number of references to a CTE clause exceeds this threshold, Hive will materialize it\n" +
         "before executing the main query block. -1 will disable this feature."),
 
+    HIVE_OPTIMIZE_REWRITE_COUNTDISTINCT_ENABLED("hive.optimize.sketches.rewrite.countdistintct.enabled", false,
+        "Enables to rewrite COUNT(DISTINCT(X)) queries to be rewritten to use sketch functions."),
+
+    HIVE_OPTIMIZE_REWRITE_COUNT_DISTINCT_SKETCHCLASS("hive.optimize.sketches.rewrite.countdistintct.sketchclass", "hll",
```

Comment:
actually I'm not 100% satisified with the "sketch class" name; but it was the best so far I've came up with. Do you have any suggestion - or it's good enough?

some alternatives I was considering:
* sketch family - I think this is too wide
* sketch framework
* sketch class
* sketch type
* ?

Reply:
What about simply `sketch`? Or `sketch type` I guess?
Family may be confusing because in their documentation they associate families with how they are commonly used, so it seems wider indeed.
http://datasketches.apache.org/docs/Architecture/SketchFeaturesMatrix.html.

Reply:
I've changed it to `sketch type` for now.

note that I'm reluctant to use the word "sketch" because we already have that in the conf's name...
I think we might consider removing the "sketches" keyword from the conf keys; that way 'sketch' could be used

* `hive.optimize.bi.rewrite.enabled`
  * or..simply: `hive.optimize.bi.enabled`  ?
* `hive.optimize.bi.rewrite.countdistintct.enabled`
* `hive.optimize.bi.rewrite.countdistintct.sketch`

now that I've written these down; I kinda like them better than the one with the "sketches" keyword in them - what do you think?

Reply:
Yes, agree, that's better.
I also like the shorter version for the main config more: `hive.optimize.bi.enabled`